### PR TITLE
[Feature] Github Issues Integration

### PR DESCRIPTION
**Pull request by davidma415**
### Problem
We want the ability for anybody to add authors into the bookstore. This means that this back-end should be able to handle webhooks given when an issue is created, edited, or deleted. In addition, we want to create a rake task that will populate the repo with some issues.

### Solution
Used Octokit in the rake task to make calls to the Github API and create issues. Added in routes and used Github Webhooks gem to handle the payloads given from the webhooks. This will allow people to create, edit, or delete issues through the repo and thus update our database for the bookstore.

### CODE REVIEW

File: app/controllers/github_webhooks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  def github_issues(payload)
+    # Get a random publishing house so we can use it to create a book.
+    randomPublisher = PublishingHouse.order('RANDOM()').first
+    case payload['action']
+    when "opened"
+      @author = Author.new(name: payload['issue']['title'], biography: payload['issue']['body'])
+
+      @book = Book.new(title: Faker::Lorem.sentence(word_count: 3), author: @author, publisher: randomPublisher, price: 24.20)
+
+      @author.save
```
Comment (ifellinaholeonce) [inline app/controllers/github_webhooks_controller.rb]: You're never checking the success or failure of saving your records. We probably want to send a different response to the frontend if the record ever fails to save.

You might also want to wrap both of these saves in a database transaction. As it stands, if an author successfully saves but the book fails, we will be left with an author with no books.

File: app/controllers/github_webhooks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    when "deleted"
+      @author = Author.find_by name: payload['issue']['title']
+      # Destroy object and associated items
+      # The reload is needed to update the author's children, and then we can records associated with the author.
+      @author.reload
+      @author.destroy
+    when "edited"
+      if payload['changes']['title']
+        @author = Author.find_by name: payload['changes']['title']['from']
```
Comment (ifellinaholeonce) [inline app/controllers/github_webhooks_controller.rb]: Again, you might want to let the frontend know if a webhook is received with an "edited" status but no author is able to be found.

File: config/application.rb [hunk 3]
```diff
@@ -21,6 +21,7 @@ module BookstoreApi
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 5.2
+    config.middleware.use ActionDispatch::Flash
```
Comment (ifellinaholeonce) [inline config/application.rb]: Why was this added?

File: config/environment.rb [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Load the Rails application.
 require_relative 'application'
 
+# Load the app's custom environment variables here, so that they are loaded before environments/*.rb
```
Comment (ifellinaholeonce) [inline config/environment.rb]: Why was it important to load them before the other environment variables?

File: app/controllers/github_webhooks_controller.rb (hunk 1 above)
Comment (davidma415) [reply app/controllers/github_webhooks_controller.rb]: Great point! When I was testing it, I thought that you would not be able to make an issue without a title, but you are able to make one without a description. Thus, I thought that it wasn't necessary.  Definitely makes sense though, will add!

File: config/application.rb (hunk 3 above)
Comment (davidma415) [reply config/application.rb]: I discovered that when using Rails in API mode, it removes a lot of the middleware. This was removed and I was running into an error when trying to access my controller functions. Is there another way to get around this?

File: config/environment.rb (hunk 4 above)
Comment (davidma415) [reply config/environment.rb]: In this case, not sure if it would matter since I don't have any other environment variables. I changed the logic for this to just set the only environment variable in `environment.rb` instead, as this removes an extra unneeded file.
